api/pkg/observability/deployment/deployment.go

```diff
@@ -0,0 +1,428 @@
+		InferenceSchema: data.ModelSchemaSpec,
+		ObservationSinks: []ObservationSink{
+			{
+				Type: Arize,
+				Config: ArizeSink{
+					APIKey:   c.consumerConfig.ArizeSink.APIKey,
+					SpaceKey: c.consumerConfig.ArizeSink.SpaceKey,
+				},
+			},
+			{
+				Type: BQ,
+				Config: BigQuerySink{
+					Project: c.consumerConfig.BigQuerySink.Project,
+					Dataset: c.consumerConfig.BigQuerySink.Dataset,
+					TTLDays: c.consumerConfig.BigQuerySink.TTLDays,
+				},
+			},
+		},
+		ObservationSource: &ObserVationSource{
+			Type: Kafka,
+			Config: &KafkaSource{
+				Topic:                    data.TopicSource,
+				BootstrapServers:         c.consumerConfig.KafkaConsumer.Brokers,
+				GroupID:                  c.consumerConfig.KafkaConsumer.GroupID,
+				BatchSize:                c.consumerConfig.KafkaConsumer.BatchSize,
+				AdditionalConsumerConfig: c.consumerConfig.KafkaConsumer.AdditionalConsumerConfig,
+			},
+		},
+	}
+	consumerCfgStr, err := yaml.Marshal(consumerCfg)
+	if err != nil {
+		return nil, err
+	}
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.getSecretName(data),
+			Namespace: data.Namespace,
+			Labels:    data.Metadata.ToLabel(),
+		},
+		StringData: map[string]string{
+			"config.yaml": string(consumerCfgStr),
+		},
+	}, nil
+}
+
+func (c *deployer) applyDeployment(ctx context.Context, data *models.WorkerData, secretName string) (*appsv1.Deployment, error) {
+	appV1 := c.kubeClient.AppsV1()
+	deploymentName := c.getDeploymentName(data)
+	deploymentV1 := appV1.Deployments(data.Namespace)
+
+	applyDeploymentFunc := func(data *models.WorkerData, secretName string, isExistingDeployment bool) (*appsv1.Deployment, error) {
+		deployment, err := c.createDeploymentSpec(ctx, data, secretName)
+		if err != nil {
+			return nil, err
+		}
+		if isExistingDeployment {
+			return deploymentV1.Update(ctx, deployment, metav1.UpdateOptions{})
+		}
+		return deploymentV1.Create(ctx, deployment, metav1.CreateOptions{})
+	}
+	_, err := deploymentV1.Get(ctx, deploymentName, metav1.GetOptions{})
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return nil, err
+		}
+		return applyDeploymentFunc(data, secretName, false)
+	}
+
+	return applyDeploymentFunc(data, secretName, true)
+}
+
+func (c *deployer) createDeploymentSpec(ctx context.Context, data *models.WorkerData, secretName string) (*appsv1.Deployment, error) {
+	labels := data.Metadata.ToLabel()
+	cfgVolName := "config-volume"
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.getDeploymentName(data),
+			Namespace: data.Namespace,
+			Labels:    labels,
+			Annotations: map[string]string{
+				PublisherRevisionAnnotationKey: strconv.Itoa(data.Revision),
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{},
+			Replicas: &c.consumerConfig.Replicas,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "worker",
```

Comment:
It seems that neither Image nor Name is updated?

Reply:
oops thanks for catching this, will update this